cmta/src/main/scala/com/lunatech/cmt/admin/command/New.scala

```diff
@@ -0,0 +1,137 @@
+package com.lunatech.cmt.admin.command
+
+import caseapp.{AppName, CommandName, ExtraName, HelpMessage, RemainingArgs, ValueDescription}
+import com.lunatech.cmt.{CmtError, printResult}
+import com.lunatech.cmt.admin.Domain.{ConfigurationFile, CourseTemplate}
+import com.lunatech.cmt.client.command.Executable
+import com.lunatech.cmt.core.validation.Validatable
+import com.lunatech.cmt.admin.cli.ArgParsers.{configurationFileArgParser, courseTemplateArgParser}
+import com.lunatech.cmt.client.Configuration
+import com.lunatech.cmt.client.cli.CmtcCommand
+import com.lunatech.cmt.Domain.InstallationSource.*
+import com.lunatech.cmt.Helpers.ignoreProcessStdOutStdErr
+import sbt.io.IO as sbtio
+import sbt.io.syntax.*
+import com.lunatech.cmt.*
+
+import cats.syntax.either.*
+
+import java.io.FileFilter
+import sys.process.*
+import util.{Try, Success, Failure}
+
+object New:
+
+  private final case class TagSet(tags: Vector[String])
+
+  @AppName("new")
+  @CommandName("new")
+  @HelpMessage(
+    "Create a new course from an existing course template in a Github repository - by default the `lunatech-labs` organisation is used.")
+  final case class Options(
+      @ExtraName("t")
+      @ValueDescription(
+        "the template course to use - provide in the format 'organisation/project' or just 'project' if the project is in the lunatech-labs organisation on Github")
+      template: CourseTemplate,
+      @ExtraName("c")
+      @ValueDescription("The (optional) configuration file to use during processing of the command")
+      @HelpMessage(
+        "if not specified will default to the config file present in the directory provided by the --main-repository argument")
+      maybeConfigFile: Option[ConfigurationFile] = None)
+
+  given Validatable[New.Options] with
+    extension (options: New.Options)
+      def validated(): Either[CmtError, New.Options] =
+        Right(options)
+  end given
+
+  given Executable[New.Options] with
+    extension (options: New.Options)
+      def execute(configuration: Configuration): Either[CmtError, String] =
+        for {
+          // list the contents of the ~/Courses directory, if there's anything already matching the
+          // name then get the count so we can append to the name and prevent conflicts
+          template <- options.template.value
+          existingFilesWithSameName = sbtio.listFiles(
+            configuration.coursesDirectory.value,
+            new FileFilter {
+              override def accept(file: File): Boolean =
+                file.name.startsWith(template.project)
+            })
+          discriminator = if (existingFilesWithSameName.size > 0) s"-${existingFilesWithSameName.size}" else ""
+          targetDirectoryName = s"${template.project}$discriminator"
+          newRepo <- newCmtRepoFromGithubProject(template, targetDirectoryName, configuration)
+        } yield newRepo
```

Comment:
Maybe break that `targetDirectoryName` logic out into its own function?

```suggestion
          template <- options.template.value
          targetDirectoryName = createTargetDirectoryName()
          newRepo <- newCmtRepoFromGithubProject(template, targetDirectoryName, configuration)
        } yield newRepo

      private def createTargetDirectoryName(): String = {
        val existingFilesWithSameName = sbtio.listFiles(
          configuration.coursesDirectory.value,
          new FileFilter {
            override def accept(file: File): Boolean =
              file.name.startsWith(template.project)
          })
        val discriminator = if (existingFilesWithSameName.size > 0) s"-${existingFilesWithSameName.size}" else ""
        s"${template.project}$discriminator"
      }
```